apps/full-site-editing/full-site-editing-plugin/full-site-editing/templates/class-wp-template-inserter.php

```diff
@@ -313,37 +313,25 @@ public function insert_default_pages() {
 
 		$api_response = json_decode( wp_remote_retrieve_body( $response ), true );
 
-		$about_page_content   = '';
-		$contact_page_content = '';
+		// Convert templates response to [ slug => content ] pairs to extract required content more easily.
+		$template_content_by_slug = wp_list_pluck( $api_response['templates'], 'content', 'slug' );
 
-		/*
-		 * Array of returned templates is not keyed by name, so we have to access it directly like this.
-		 * About page is at position 6 in the array, and Contact page at 1.
-		 */
-		if ( ! empty( $api_response['templates'][6]['content'] ) ) {
-			$about_page_content = $api_response['templates'][6]['content'];
-		}
-
-		if ( ! empty( $api_response['templates'][1]['content'] ) ) {
-			$contact_page_content = $api_response['templates'][1]['content'];
-		}
-
-		if ( empty( get_page_by_title( 'About' ) ) ) {
+		if ( empty( get_page_by_title( 'About' ) ) && ! empty( $template_content_by_slug['about'] ) ) {
 			wp_insert_post(
 				[
 					'post_title'   => _x( 'About', 'Default page title', 'full-site-editing' ),
-					'post_content' => $about_page_content,
+					'post_content' => $template_content_by_slug['about'],
```

Comment:
Are there any issues with localization and template slugs? These stay the same regardless of language?

Reply:
My assumption was that that was the rationale behind adding them (other parts of the response like title and content will be translated). cc-ing @obenland to confirm

Reply:
Actually no need - I just verified that slug remains unchanged for other languages too.